addon/components/power-calendar-range.js

```diff
@@ -66,8 +67,19 @@ export default CalendarComponent.extend({
 
   // Actions
   actions: {
-    select(day, calendar, e) {
-      let range = this._buildRange(day);
+    select({ date }, calendar, e) {
+      assert(
+        'date must be either a Date, or a Range', 
+        date && (date.hasOwnProperty('start') || date.hasOwnProperty('end') || date instanceof Date)
```

Comment:
nit pick: I'd put the `date instanceof Date` the first because that is the most common use-case.

Reply:
On a second read it doesn't really matter because it's on the assert and it will be gone in production.